diff --git video_embed_field/src/Plugin/Field/FieldFormatter/Video.php video_embed_field/src/Plugin/Field/FieldFormatter/Video.php
index 84a20c8..361207a 100644
--- video_embed_field/src/Plugin/Field/FieldFormatter/Video.php
+++ video_embed_field/src/Plugin/Field/FieldFormatter/Video.php
@@ -126,7 +126,7 @@ class Video extends FormatterBase implements ContainerFactoryPluginInterface {
       'responsive' => TRUE,
       'width' => '854',
       'height' => '480',
-      'autoplay' => FALSE,
+      'autoplay' => TRUE,
     ];
   }
 
